### ISSUE TITLE
Feature Request: Enhanced Markdown Export with Linked References for Obsidian Compatibility

### ISSUE THREAD

**Issue by SchneiderSam** [erykjj/jwlmanager#194]
Hi @erykjj 

Again: I'd like to express my gratitude for your incredible work on JWL Manager! The Markdown export feature works seamlessly and has greatly enhanced my workflow, especially when importing JWL notes into Obsidian. **I truly appreciate all the effort you put into making this tool so effective**.

As I’ve been using JWL Manager for exporting extensive JWL notes, I’ve noticed a **potential enhancement** that could make working with these JWL notes in Obsidian even more powerful. Currently, the exported notes—for example in the Study Bible—appear as numerous orphan notes within Obsidian's GraphView, which makes it challenging to establish clear connections between them.

### Feature Suggestion:
I’ve noticed that during the export process, two essential metadata elements are included: the *Publication* and, in cases like Study Bible notes, the *Bible verse*. Would it be possible to modify the Markdown export so that these metadata elements (Heading and Publication) are enclosed in double brackets (e.g., `heading: [[2. Mose 22_18]]` or `publication: [[Rbi8-X]]`)? This would effectively create links to those references within Obsidian. I can only imagine the ingenious connections that are created on the notes. There would even be an interesting connection for the colors. Most people always use the colors in a certain context (e.g., `color: [[3]]`).

If you only want this to be optional, you could also activate or deactivate an option when exporting whether you want to have these double brackets or not.

### Possible Challenges:

**Special Characters in Bible Verses:**
Certain characters, like colons, aren't allowed in Obsidian titles. For instance, "2. Mose 22:18" could be formatted as:

- `[[2-Mose-22-18]]`: Replace special characters with hyphens for compatibility.
- `[[2. Mose 22_18]]`: Use an underscore instead of a colon for chapter and verse. (my favorite)
- `[[2-Mose-(22-18)]]`: Enclose chapter and verse in parentheses with hyphens.
- `[[02022018]]`: Like in the JWL Link `https://www.jw.org/finder?wtlocale=2&pub=nwt&bible=**02022018**` (not so easy to read)

**Standardized Book Formatting:**
For publications, formats like nwt-X could simply be converted to [[nwt-X]], ensuring compatibility across notes.

### Benefits:
1. **Improved Organization:** Many orphan notes would automatically become connected, allowing for a much cleaner and organized GraphView in Obsidian or Logseq or other Markdown Tools.
2. **Enhanced Navigation:** By linking notes to Bible verses or publications, users can navigate more intuitively between related notes.
3. **Powerful Knowledge Map:** The ability to visualize and interact with these connections transforms the exported notes into a truly cohesive JWL knowledge map.

Thank you very much for considering this feature! I believe this addition would provide tremendous value to users who work with JWL Manager and Obsidian together. I would be more than happy to provide further details or test any potential implementations.

**Comment by erykjj** [comment]
@SchneiderSam
This could easily be implemented, but I need to know a few things:

1. the [[doube brackets]] are what you use for linking to other notes in Obsidian?
2. would there be an issue if the [[double brackets]] were always included?
  1. would it detract in any way from the usefulness of the metadata section?
  2. how would it affect other markdown viewer/editor applications?
3. which ones would most benefit from this? color, language, publication, document?
4. for Bible notes, I could put `01001001` (Genesis 1:1) in the document attribute. would that work/help?
5. the title attribute, if in square brackets, cannot contain special characters and needs to be modified? I could make it like the filename. would that work?

**Comment by erykjj** [comment]
Please test [v6.0.0-beta3](https://github.com/erykjj/jwlmanager/releases/tag/v6.0.0-beta3), which tries to implement some of the above suggestions.

Also, please test/report on the theme toggle and translations ;-)

**Comment by SchneiderSam** [comment]
> Please test [v6.0.0-beta3](https://github.com/erykjj/jwlmanager/releases/tag/v6.0.0-beta3), which tries to implement some of the above suggestions.

@erykjj  Works fine. Wow 🥳🥳

Two Tests:

**1. theme toggle (light/dark mode)**

works great and looks good



**2. Markdown export:**

> the [[doube brackets]] are what you use for linking to other notes in Obsidian?

Yes, https://help.obsidian.md/Linking+notes+and+files/Internal+links

> would there be an issue if the [[double brackets]] were always included?

I have to test, but it should work fine. Pls give me a day...

> would it detract in any way from the usefulness of the metadata section? 

I don't think so. But that is a very subjective opinion. 😅

> how would it affect other markdown viewer/editor applications?

It's not a Markdown Basic Syntax. See https://www.markdownguide.org/basic-syntax/#links
But other Tools like https://roamresearch.com/ and https://docs.logseq.com/#/page/aliases%20and%20external%20links are using this system with dubble Brackets also.

> which ones would most benefit from this? color, language, publication, document?

color, publication, document

> for Bible notes, I could put 01001001 (Genesis 1:1) in the document attribute. would that work/help?

It's ok. But can you add a separate Meta field called `aliases:` Genesis 1:1

So in this case we have two Informations:

```
---
...
document: 01001001
aliases:
  - Genesis 1:1
---
```
This would simplify the search and we have both information.
Please see: https://help.obsidian.md/Linking+notes+and+files/Aliases

> the title attribute, if in square brackets, cannot contain special characters and needs to be modified? I could make it like the filename. would that work?

Let me check that tomorrow. Have to check this further.

**Comment by SchneiderSam** [comment]
Ok that doesn't work... i have To Check further...
![image](https://github.com/user-attachments/assets/ab9ab8c1-5225-4909-bd2a-3eba84101e4e)

![image](https://github.com/user-attachments/assets/952f4690-a6ff-49b2-83ce-cba1e0aadb70)

The Double Brackets Don't work on meta field...

**Comment by SchneiderSam** [comment]
I found a Solution... With These options it works well:

> Internal links in Text and List type properties must be surrounded with quotes. Obsidian will automatically add these if you manually enter internal links into properties, but be careful to add them when using templating plugins.

`document: "[[202017367]]"`

https://help.obsidian.md/Editing+and+formatting/Properties

**Comment by SchneiderSam** [comment]
I have created a test vault for myself and created a new export. Please excuse the many posts, but my brain has not rested during the night...

I'll summarize briefly here. This is what I currently see with beta3

View Mode:
![image](https://github.com/user-attachments/assets/fd2caa86-f1e1-4221-915e-cf08659635db)

Source Mode:
![image](https://github.com/user-attachments/assets/f7719af5-bc95-4e3d-b916-9d31af4ab1e7)


Better would be:
![image](https://github.com/user-attachments/assets/7ea5e953-326c-4b72-8fb8-d9298e6f714c)
![image](https://github.com/user-attachments/assets/b95e5a79-7fa6-4d02-bbc9-f55d1ae4ddf3)


But: I'm not sure whether it makes sense to display the title as a link (e.g. `title: "[[lebendig]]"`). You are welcome to leave it, but perhaps a further development of the export would be: a selective choice where you want to have links everywhere. But that's really not necessary for this version. The way you have planned it is really good.


How is the META `heading` created? This is not the case for every publication. Only with the Bible. Correct?  That's fine, in the example `66019020` = `Revelation 19:20`

So I could create notes (e.g. `66019020`) for myself later and then define an aliases (e.g. `Revelation 19:20`). Perfect. Please forget what I wrote in the field above, that makes no sense. On the subject of aliases, which I mentioned here: https://github.com/erykjj/jwlmanager/issues/194#issuecomment-2460774369 😅

**Comment by SchneiderSam** [comment]
Comment updated: https://github.com/erykjj/jwlmanager/issues/194#issuecomment-2460774369

**Comment by erykjj** [comment]
Thanks for all that, @SchneiderSam !
I will remove [[]] from the title and add "" around the remaining [[]].

**Comment by erykjj** [comment]
Should each individual tag also be in this format: `"[[tag]]"`?

**Comment by erykjj** [comment]
And would `"[[nwtsty]]"` be better than `"[[nwtsty-E]]"` (since there is a separate attribute for language)?

**Comment by SchneiderSam** [comment]
> And would `"[[nwtsty]]"` be better than `"[[nwtsty-E]]"` (since there is a separate attribute for language)?

Good question. I think both options are ok. I don't currently have any special requirements.

**Comment by SchneiderSam** [comment]
> Should each individual tag also be in this format: `"[[tag]]"`?

This is not necessary for obsidian. It automatically recognizes this as a tag. I have reformulated all my JWL tags. Example: `#MoC/love`

It Works great with brackets and Quotes

**Comment by erykjj** [comment]
[v6.0.0-beta4](https://github.com/erykjj/jwlmanager/releases/tag/v6.0.0-beta4) is out. Please report if the attributes/properties in the markdown export are properly formatted (for Obsidian to handle) and if the cross-linking works ;-)

**Comment by SchneiderSam** [comment]
> [v6.0.0-beta4](https://github.com/erykjj/jwlmanager/releases/tag/v6.0.0-beta4) is out. Please report if the attributes/properties in the markdown export are properly formatted (for Obsidian to handle) and if the cross-linking works ;-)

I'am on a Trip. But i will test it later this afternoon or tomorrow

**Comment by SchneiderSam** [comment]
@erykjj wow it works very well. All JWL files got perfect links.

But one Problem:

This is indirectly related to the new feature, I think we're just noticing it now.
As soon as a colon appears again after `heading:` an error is displayed.

In the example `heading: Teil 2:`.
![image](https://github.com/user-attachments/assets/04a0e4cc-ec60-4cc1-98b0-d998e4df219b)

If the colon is gone, then it works and the property box is displayed correctly.
![image](https://github.com/user-attachments/assets/b4bda90e-7a79-4a57-9d15-11d2166253ed)

![image](https://github.com/user-attachments/assets/5d8bf921-3fb1-4ef8-b8e5-2127d7783a72)

You can see the error by the fact that Obsidian displays an error (it is red).

## Solution
Can you put quotes around the heading content? Like this:
![image](https://github.com/user-attachments/assets/133bb5a7-3654-4142-835e-4cc9ae5cb4b1)

Then t works fine:
![image](https://github.com/user-attachments/assets/d8150cae-05f6-4ec5-aa15-ec59b560ba93)


After that for me everything works amazing. THANKS!!

**Comment by erykjj** [comment]
OK, I can add the quotation marks around the heading.
I found a similar behavior with tags that contain special characters. How would tags behave if they also were all in quotations marks:

```
- tags:
  - "test"
  - "*"
```
Would that still work well for you??

**Comment by SchneiderSam** [comment]
> Would that still work well for you??

it works:
![image](https://github.com/user-attachments/assets/144b512f-c10a-4d8d-9101-6d26ac97c49c)


but every tag need two quotes. Thanks :-)

**Comment by erykjj** [comment]
> every tag need two quotes

Not sure I understand. Do you mean like this?
```
tags:
  - ""tag""
```

**Comment by SchneiderSam** [comment]
> > every tag need two quotes
> 
> Not sure I understand. Do you mean like this?
> 
> ```
> tags:
>   - ""tag""
> ```

No. Sorry, I didn't make myself clear. I meant that if there is more than one tag, then each tag needs the quotation marks. So that's correct:


```
tags:
 - "tag 1"
 - "tag 2"
 - "tag 3"
```

**Comment by SchneiderSam** [comment]
@erykjj What do you think? Should we also put quotation marks around “title:” and “heading:” to be on the safe side?

In my test, I simply added a special character to the title. If we don't use quotation marks here, the error will also appear. This way we could prevent this if a special character should “get lost”.

**Comment by erykjj** [comment]
I am ready to realease. Any final touches/suggestions on your part @SchneiderSam ?

**Comment by SchneiderSam** [comment]
> I am ready to realease. Any final touches/suggestions on your part @SchneiderSam ?

All fine 😁💪🫶

**Comment by erykjj** [comment]
Implemented in [v6.0.0](https://github.com/erykjj/jwlmanager/releases/tag/v6.0.0)
